lib/_http_client.js

```diff
@@ -322,7 +322,7 @@ function socketOnData(d) {
     var bodyHead = d.slice(bytesParsed, d.length);
 
     var eventName = req.method === 'CONNECT' ? 'connect' : 'upgrade';
-    if (req.listenerCount(eventName) > 0) {
+    if (EventEmitter.listenerCount(req, eventName) > 0) {
```

Comment:
`req.listenerCount` _should_ be fine, unless userland code is punching the `_httpMessage` property.